### PR TITLE
value_counts support

### PR DESCRIPTION
**Pull request by maartenbreddels**
This is a start for value_counts, only works/tested for 'integer' data. 
TODO:
 - [x] test with strings
 - [x] test with float values 0.1, etc
 - [x] consider refactoring/renaming label_encoding/catergorise
 - [ ] test the changes in catergorise

### CODE REVIEW

File: packages/vaex-core/vaex/expression.py [hunk 1]
```diff
@@ -276,6 +276,61 @@ def max(self, binby=[], limits=None, shape=default_shape, selection=False, delay
         kwargs['expression'] = self.expression
         return self.ds.max(**kwargs)
 
+    def value_counts(self, dropna=False, ascending=False):
+        """Computes counts of unique values
+         WARNING:
+          * If the expression/column is not categorical, it will be converted on the fly
+          * dropna is False by default, it is True by default in pandas
+         :param dropna: when True, it will not report the missing values
```
Comment (JovanVeljanoski) [inline packages/vaex-core/vaex/expression.py]: Formatting is a little of here.

File: packages/vaex-core/vaex/expression.py [hunk 2]
```diff
@@ -276,6 +276,61 @@ def max(self, binby=[], limits=None, shape=default_shape, selection=False, delay
...
+        """
+        from pandas import Series
+        df = self.ds
+        # whatever we do, it's gonna be similar to ds.categorise, so better
+        # to use that path
+        if not df.iscategory(self.expression):
+            df = df.copy()
+            # consider using label_encode.. which seems like a wrong name actually
+            # ds.categorize(self.expression)
+            df = df.label_encode(self.expression)
```
Comment (JovanVeljanoski) [inline packages/vaex-core/vaex/expression.py]: Should this be `df.ordinal_encode` now?

File: packages/vaex-core/vaex/expression.py [hunk 3]
```diff
@@ -276,6 +276,61 @@ def max(self, binby=[], limits=None, shape=default_shape, selection=False, delay
...
+         :param dropna: when True, it will not report the missing values
+        :param ascending: when False (default) it will report the most frequent occuring item first
+        :returns: Pandas series containing the counts
+        """
+        from pandas import Series
+        df = self.ds
+        # whatever we do, it's gonna be similar to ds.categorise, so better
+        # to use that path
+        if not df.iscategory(self.expression):
+            df = df.copy()
```
Comment (JovanVeljanoski) [inline packages/vaex-core/vaex/expression.py]: I don't understand what is the purpose of this line.. can you please explain?

File: packages/vaex-core/vaex/expression.py [hunk 4]
```diff
@@ -276,6 +276,61 @@ def max(self, binby=[], limits=None, shape=default_shape, selection=False, delay
...
+            if np.any(nan_mask):
+                ok = ok & ~nan_mask
+        if np.ma.isMaskedArray(index):
+            ok = ok & ~np.ma.getmaskarray(index)
+        counts = counts[ok]
+        index = index[ok]
+        return Series(counts, index=index)
+
+    def unique(self):
+        return self.ds.unique(self.expression)
```
Comment (JovanVeljanoski) [inline packages/vaex-core/vaex/expression.py]: Should this be also renamed to `self.df.unique` at this point?

File: tests/category_test.py [hunk 5]
```diff
@@ -5,40 +5,49 @@
...
+    ds0 = vaex.from_arrays(colors=['red', 'green', 'blue', 'green'])
+    ds = ds0.label_encode('colors')#, ['red', 'green'], inplace=True)
+    assert ds.iscategory('colors')
+    assert ds.limits('colors', shape=128) == ([-0.5, 2.5], 3)
 
-	ds = ds0.label_encode('colors', values=['red', 'green'])
-	assert ds.iscategory('colors')
-	assert ds.limits('colors', shape=128) == ([-0.5, 1.5], 2)
-	assert ds.data.colors.tolist() == [0, 1, None, 1]
+    ds = ds0.label_encode('colors', values=['red', 'green'])
```
Comment (JovanVeljanoski) [inline tests/category_test.py]: perhaps re-write as `ordinal_encode` if I understand this correctly...?

File: tests/category_test.py [hunk 6]
```diff
@@ -5,40 +5,49 @@
...
+    # 	assert ds.iscategory('colors', values=['red', 'orange'])
 
 def test_count_cat():
-	ds0 = vaex.from_arrays(colors=['red', 'green', 'blue', 'green'], counts=[1, 2, 3, 4])
-	ds0 = vaex.from_arrays(colors=['red', 'green', 'blue', 'green'], names=['apple', 'apple', 'berry', 'apple'])
-	ds = ds0.label_encode(ds0.colors)
-	ds = ds0.label_encode(ds0.names)	
+    ds0 = vaex.from_arrays(colors=['red', 'green', 'blue', 'green'], counts=[1, 2, 3, 4])
+    ds0 = vaex.from_arrays(colors=['red', 'green', 'blue', 'green'], names=['apple', 'apple', 'berry', 'apple'])
+    ds = ds0.label_encode(ds0.colors)
```
Comment (JovanVeljanoski) [inline tests/category_test.py]: Here and the line below `ordinal_encode` instead of `label_encode`.

File: tests/category_test.py [hunk 7]
```diff
@@ -5,40 +5,49 @@
...
+    assert ds0.iscategory(ds0.c)
+    assert ds0.category_labels(ds0.c) == ['a', 'b', 'c', 'd']
+    assert ds0.category_count(ds0.c) == 4
+
+def test_cat_missing_values():
+    colors = ['red', 'green', 'blue', 'green', 'MISSING']
+    mask   = [False, False,   False,   False,  True]
+    colors = np.ma.array(colors, mask=mask)
+    ds0 = vaex.from_arrays(colors=colors)
+    ds = ds0.label_encode('colors', ['red', 'green', 'blue'])
```
Comment (JovanVeljanoski) [inline tests/category_test.py]: One more time the same comment about `label_encode` to `ordinal_encode`.

File: packages/vaex-core/vaex/expression.py (hunk 3 above)
Comment (maartenbreddels) [reply packages/vaex-core/vaex/expression.py]: I thought inplace=True was default, which is not

File: packages/vaex-core/vaex/expression.py (hunk 4 above)
Comment (maartenbreddels) [reply packages/vaex-core/vaex/expression.py]: yes, lets do that separately though, not in this PR.

**Pull request merged**